phoenix-core/src/main/java/org/apache/phoenix/compile/JoinCompiler.java

```diff
@@ -761,13 +761,15 @@ public Expression compilePostFilterExpression(StatementContext context) throws S
         }
 
         public SelectStatement getAsSubquery(List<OrderByNode> orderBy) throws SQLException {
-            if (isSubselect())
-                return SubselectRewriter.applyOrderBy(
+            if (isSubselect()) {
+                return SubselectRewriter.applyOrderByAndPostFilters(
                         SubselectRewriter.applyPostFilters(subselect, preFilters, tableNode.getAlias()),
                         orderBy,
                         tableNode.getAlias(),
-                        tableNode);
-
+                        postFilters);
+            }
+            //for table, postFilters is empty , because it can safely pushed down as preFilters.
```

Comment:
If there are ```postFilters``` will they always be pushed down into the sub query?

Reply:
Yes, the postFilters which are pushed down to JoinCompiler.Table could always be pushed down into the subquery, I would omit the useless code to make it more clear.
By contrast, the postFilters of JoinCompiler.JoinTable should be filtered after the join is completed.